lessons/04-client-jquery-ajax/README.md

```diff
@@ -1,517 +1,131 @@
+If you ever run into errors connecting to your database, ensure the `mongod` service (mongo daemon) is still running, and restart it if necessary.
+
+While the server is running in the background you can exit it safely by running:
+```sh
+sudo service mongod stop
+```
+
+While the database is running you can access it directly through your terminal using the `mongo` command. This will connect on the default Mongo port (27017) and allow you to issue MongoDB commands directly:
+```sh
+$ mongo
+MongoDB shell version: 2.6.6
+connecting to: test
+>
+```
+You can see all the databases you have on your machine by typing
+```sh
+> show dbs
+admin  (empty)
+local  0.078GB
+```
+Let's create a new database and a new entry in that database:
+```sh
+> use test
+switched to db test
+> db.users.insert({'name':'alice'})
+> db.users.find()
+{ "_id" : ObjectId("51007865e481634f390b162f"), "name" : "alice" }
+> db.users.insert({'name': 'bob', 'grade': 'A', 'assignments':[{1: 'A', 2: 'B'}]})
+> db.users.find()
+{ "_id" : ObjectId("51007865e481634f390b162f"), "name" : "alice" }
+{ "_id" : ObjectId("510078bee481634f390b1630"), "name" : "bob", "grade" : "A",
+"assignments" : [ { "1" : "A", "2" : "B" } ] }
+> show dbs
+admin (empty)
+local 0.078GB
+test  0.203125GB
+```
+Mongo creates a database for us as soon as we start inserting items into it. It stores data in what's known as a `collection`. So in our case, users would be a collection. Items within a collection don't have to be consistent with each other (alice only has a name while bob has a name and a grade).
+
+We can also delete specific items, or a whole collection:
+```
+> db.users.remove({'name': 'alice'})
+> db.users.find()
+{ "_id" : ObjectId("510078bee481634f390b1630"), "name" : "bob", "grade" : "A",
+"assignments" : [ { "grade" : "A", "grade" : "B" } ] }
+> db.users.drop()
+true
+```
+Modify items:
+```sh
+> db.users.update({'name': 'bob'}, {$set: {'class': 2013}})
+> db.users.find()
+{ "_id" : ObjectId("510078bee481634f390b1630"), "assignments" : [ { "grade" : "A", "grade" : "B" } ],
+"class" : 2013, "grade" : "A", "name" : "bob" }
+```
+And search for items:
+```sh
+> db.users.find({'grade': 'A'})
+{ "_id" : ObjectId("510078bee481634f390b1630"), "assignments" : [ { "grade" : "A", "grade" : "B" } ],
+"class" : 2013, "grade" : "A", "name" : "bob" }
+```
+There are loads more Mongo commands that can be found through the [documentation](http://docs.mongodb.org/manual/).
+## Mongoose
+[Mongoose](http://mongoosejs.com/) is a javascript wrapper for MongoDB that allows us to save javascript objects into our database without having to deal with the underlying Mongo commands. Install it by opening up a console and typing:
+```sh
+npm install --save mongoose
+```
+You can configure Mongoose to save objects using a [schema](http://mongoosejs.com/docs/guide.html). In a schema we define what kind of data we expect an object to have.
 ```javascript
-$("h1").html("That was easy")
+var userSchema = mongoose.Schema({
```

Comment:
We may want to include a bit about connecting to a mongo database from your app using Mongoose (which may also lead to an opportunity to touch on environment variables (and perhaps [dotenv](https://www.npmjs.com/package/dotenv) for configuration)